docs_src/design/adr/0005-Service-Self-Config.md

```diff
@@ -0,0 +1,74 @@
+
+- --configProvider or -cp (the configuration provider location URL)
+- --overwrite or -o (overwrite the configuration in the configuration provider)
+- --file or -f (the configuration filename)
+- --profile or -p (the profile name of configuration)
+- --confidir (the directory where the configuration file is found)
+- --registry or -r (use the registry)
+
+The distinction of op params versus configuration will be important later in this ADR.
+
+### Configuration Initializaiton
+Going forward, each service would have a local configuration file.  On intialization of the service (aka bootstrap of the service), the service would use the configuration to populate the configuration service (Consul) with data read from the configuration file.  In this way, each service self-seeds the configuration service.
+
+Details of the proposed process are as follows:
+- Service initializes and determines whether it should get its configuration from a local configuration file or from a configuration service by way of 
+  
+  a) the -cp / --configProvider op param
+
+  b) the edgex_configuration_provider environmental variable which overrides the op param above
+  
+- If the service is to use the local file, it reads the file associated with the profile, config filename and config file directory op params or environmental variables) and does not contact the configuration service (Consul) for any configuration information.
+- If the service is to use the configuration service, then the service will:
+  
+  1) call on Consul and check the root in Consul to see if it is already populated with configuration information provided by the service.  If so, the service will load and use the Consul provided configuration for that service.
+  2) If the root in Consul is not populated with configuration information provided by the service, the service will read the local configuration file and populate Consul with the configuration contents in that file.
+
+NOTE: as the services now self seed and the profile allows for the appropriate seeding to occur in the service itself, it will no longer be necessary to have a Docker configuration file in each of the service directories (example:  https://github.com/edgexfoundry/edgex-go/blob/master/cmd/core-data/res/docker/configuration.toml).  See Consequences below.
+
+### Overrides
+Environmental variable overrides of configuration or op param values can occur in one of two ways
+- Environmental variables can override configuration values provided by Consul.  The name of the environmental variable must match the path names in Consul.  NOTE:  environment variable overrides do not get pushed into Consul.  They only override a value read from Consul on bootstrap.
+- Environmental variables can override op params.  Op params specify opreation of the service and are configuration that is not ever in Consul. 
```

Comment:
I think we should list the non-boolean op params that can be overridden with the env var names.  Do we want to include -r/--registry as override'able? It doesn't make sense to me to override -o/-overwrite.